File: src/main/java/me/nobaboy/nobaaddons/features/chatcommands/impl/party/CoordsCommand.java
package me.nobaboy.nobaaddons.features.chatcommands.impl.party;

import me.nobaboy.nobaaddons.NobaAddons;
import me.nobaboy.nobaaddons.features.chatcommands.ChatContext;
import me.nobaboy.nobaaddons.features.chatcommands.IChatCommand;
import me.nobaboy.nobaaddons.util.ChatUtils;
import net.minecraft.client.Minecraft;
import net.minecraft.client.entity.EntityPlayerSP;

public class CoordsCommand implements IChatCommand {
    @Override
    public String name() {
        return "coords";
    }

    @Override
    public void run(ChatContext ctx) {
        EntityPlayerSP player = Minecraft.getMinecraft().thePlayer;
        ChatUtils.delayedSend("pc x: " + (int) player.posX + ", y: " + (int) player.posY + ", z: " + (int) player.posZ);
    }

    @Override
    public boolean isEnabled() {
        return NobaAddons.config.partyCoordsCommand;
    }
}

File: src/main/kotlin/me/nobaboy/nobaaddons/features/chatcommands/impl/party/CoordsCommand.kt
package me.nobaboy.nobaaddons.features.chatcommands.impl.party

import me.nobaboy.nobaaddons.NobaAddons
import me.nobaboy.nobaaddons.NobaAddons.Companion.mc
import me.nobaboy.nobaaddons.features.chatcommands.ChatContext
import me.nobaboy.nobaaddons.features.chatcommands.IChatCommand
import me.nobaboy.nobaaddons.util.HypixelCommands

class CoordsCommand : IChatCommand {
    override val name: String = "coords"

    override val isEnabled: Boolean
        get() = NobaAddons.config.partyCoordsCommand

    override fun run(ctx: ChatContext) {
        val player = mc.thePlayer
        val posX = player.posX.toInt()
        val posY = player.posY.toInt()
        val posZ = player.posZ.toInt()

        HypixelCommands.partyChat("x: $posX, y: $posY, z: $posZ")
    }
}